### ISSUE TITLE
Implement Verifiable Credentials

### ISSUE THREAD

**Issue by amika-sq** [TBD54566975/web5-swift#4]
iOS needs to implement Verifiable Credentials, so that a consumer of the SDK can create/verify/etc a claim required for tbDEX

**Comment by angiejones** [comment]
> [after discussion with @amika-sq] this may not be needed. a client will only need `PresentationExchange.selectCredentials` when fulfilling the `requiredClaims` of the `offering` and providing the associated data to the `claims` of the `rfq`
> 
> good to close @amika-sq ?

Wallet also needs `PresentationExchange.satisfiesPresentationDefinition`

**Comment by angiejones** [comment]
> > Wallet also needs PresentationExchange.satisfiesPresentationDefinition
> 
> ah gotcha, assuming so we can present to the user why certain credentials didn't satisfy a PD.
> 
> if no matching creds are found, `selectCredentials` will return an empty list while `satisfiesPresentationDefinition` will return an errors object explaining why.
> 
> if 2/5 matching creds are found, `selectCredentials` will return a list size of 2 which would be insufficient to satisfy the claims requirements, while `satisfiesPresentationDefinition` would return an errors object describing which requirements failed.
> 
> makes sense, good catch thank you!

Yes, more specifically, a Wallet can use this method to find the Offerings where the customer already has VCs that will satisfy them - therefore eliminating Offerings that she doesn't qualify for.

Here's [an example](https://developer.tbd.website/docs/tbdex/wallet/managing-credentials#obtain-eligible-offerings)
